sonic_platform_base/sonic_sfp/sfputilbase.py

```diff
@@ -0,0 +1,1108 @@
+                                                        str(i2c_adapter_id),
+                                                        hex(devid)[-2:])
+
+            # If sfp device is not present on bus, Add it
+            if not os.path.exists(sysfs_sfp_i2c_client_path):
+                ret = self._add_new_sfp_device(
+                        sysfs_sfp_i2c_adapter_path, devid)
+                if ret != 0:
+                    print("Error adding sfp device")
+                    return None
+
+            sysfs_sfp_i2c_client_eeprom_path = "%s/eeprom" % sysfs_sfp_i2c_client_path
+
+        return sysfs_sfp_i2c_client_eeprom_path
+
+    # Read out any bytes from any offset
+    def _read_eeprom_specific_bytes(self, sysfsfile_eeprom, offset, num_bytes):
+        eeprom_raw = []
+        for i in range(0, num_bytes):
+            eeprom_raw.append("0x00")
+
+        try:
+            sysfsfile_eeprom.seek(offset)
+            raw = sysfsfile_eeprom.read(num_bytes)
+        except IOError:
+            print("Error: reading sysfs file %s" % sysfs_sfp_i2c_client_eeprom_path)
+            return None
+
+        try:
+            for n in range(0, num_bytes):
+                eeprom_raw[n] = hex(ord(raw[n]))[2:].zfill(2)
+        except:
+            return None
+
+        return eeprom_raw
+
+    # Read eeprom
+    def _read_eeprom_devid(self, port_num, devid, offset, num_bytes = 256):
+        sysfs_sfp_i2c_client_eeprom_path = self._get_port_eeprom_path(port_num, devid)
+
+        if not self._sfp_eeprom_present(sysfs_sfp_i2c_client_eeprom_path, offset):
+            return None
+
+        try:
+            sysfsfile_eeprom = open(sysfs_sfp_i2c_client_eeprom_path, mode="rb", buffering=0)
+        except IOError:
+            print("Error: reading sysfs file %s" % sysfs_sfp_i2c_client_eeprom_path)
+            return None
+
+        eeprom_raw = self._read_eeprom_specific_bytes(sysfsfile_eeprom, offset, num_bytes)
+
+        try:
+            sysfsfile_eeprom.close()
+        except:
+            return None
+
+        return eeprom_raw
+
+    def _is_valid_port(self, port_num):
+        if port_num >= self.port_start and port_num <= self.port_end:
+            return True
+
+        return False
+
+    def read_porttab_mappings(self, porttabfile):
+        logical = []
+        logical_to_bcm = {}
+        logical_to_physical = {}
+        physical_to_logical = {}
+        last_fp_port_index = 0
+        last_portname = ""
+        first = 1
+        port_pos_in_file = 0
+        parse_fmt_port_config_ini = False
+
+        try:
+            f = open(porttabfile)
+        except:
+            raise
+
+        parse_fmt_port_config_ini = (os.path.basename(porttabfile) == "port_config.ini")
+
+        # Read the porttab file and generate dicts
+        # with mapping for future reference.
+        #
+        # TODO: Refactor this to use the portconfig.py module that now
+        # exists as part of the sonic-config-engine package.
```

Comment:
I notice the TODO is quite reasonable, and there are lots of repeated port_config parser in code base. Can we start take the action?

Reply:
This is a good time to to make the change, because there are open PRs to begin transitioning from the port_config.ini file to a new 'platform.json' file. The logic to do the parsing and also to fallback to the old port_config.ini is all in the portconfig.py module: https://github.com/Azure/sonic-buildimage/pull/3909